### PR TITLE
[JENKINS-69304] Allow specifying a java path

### PR DESCRIPTION
**Pull request by lemeurherve**
Cf https://issues.jenkins.io/browse/JENKINS-69304 and https://github.com/jenkins-infra/helpdesk/issues/3090, this PR add a "javaPath" option.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/hudson/plugins/ec2/win/EC2WindowsLauncher.java [hunk 1]
```diff
@@ -91,10 +91,11 @@ protected void launchScript(EC2Computer computer, TaskListener listener) throws
 
             logger.println("remoting.jar sent remotely. Bootstrapping it");
 
+            final String javaPath = node.javaPath;
             final String jvmopts = node.jvmopts;
             final String remoteFS = WindowsUtil.quoteArgument(node.getRemoteFS());
             final String workDir = Util.fixEmptyAndTrim(remoteFS) != null ? remoteFS : tmpDir;
-            final String launchString = "java " + (jvmopts != null ? jvmopts : "") + " -jar " + tmpDir + AGENT_JAR + " -workDir " + workDir;
+            final String launchString = javaPath + " " + (jvmopts != null ? jvmopts : "") + " -jar " + tmpDir + AGENT_JAR + " -workDir " + workDir;
```
Comment (lemeurherve) [inline src/main/java/hudson/plugins/ec2/win/EC2WindowsLauncher.java:98]: Note: unlike the Mac & Unix versions, this launcher doesn't allow specifying command `prefix` or `suffix`

File: src/main/java/hudson/plugins/ec2/SlaveTemplate.java [hunk 2]
```diff
@@ -510,6 +520,7 @@ public SlaveTemplate(String ami, String zone, SpotConfiguration spotConfig, Stri
                 monitoring, t2Unlimited, ConnectionStrategy.backwardsCompatible(usePrivateDnsName, connectUsingPublicIp, associatePublicIp), -1);
     }
 
+    @Deprecated
```
Comment (lemeurherve) [inline src/main/java/hudson/plugins/ec2/SlaveTemplate.java:523]: Added these `@Deprecated` annotations, not completely sure they are needed.

**Comment by alecharp** [comment]
@fcojfernandez would you have time to review this? Thanks.

**Comment by lemeurherve** [comment]
Manually tested an incremental on infra.ci.jenkins.io ✅

**Comment by Hentheroot** [comment]
Hi, all very nice. But if you keep 'javaPath' empty (like me) I expect the system does a 'which java' to selfdetect the java binary.
Because now I kept it empty the launch of an EC2 failed.

**Comment by res0nance** [comment]
> Hi, all very nice. But if you keep 'javaPath' empty (like me) I expect the system does a 'which java' to selfdetect the java binary. Because now I kept it empty the launch of an EC2 failed so I was under the impression version 2.0.0 had a bug so I rolled back.

Has the 2.0.1 version resolved this for you?

**Comment by Hentheroot** [comment]
> Has the 2.0.1 version resolved this for you?

Yes it did. Thanks for the quick fix.

**Pull request merged**